### PR TITLE
Manually Update System.Text.Json from 8.0.0 to 8.0.3

### PR DESCRIPTION
**Pull request by surayya-MS**
https://dev.azure.com/devdiv/DevDiv/_git/VS/pullrequest/543265?_a=files

### CODE REVIEW

File: eng/Version.Details.xml [hunk 1]
```diff
@@ -53,7 +53,7 @@
       <Uri>https://dev.azure.com/dnceng/internal/_git/dotnet-runtime</Uri>
       <Sha>5535e31a712343a63f5d7d796cd874e563e5ac14</Sha>
     </Dependency>
-    <Dependency Name="System.Text.Json" Version="8.0.0">
+    <Dependency Name="System.Text.Json" Version="8.0.3">
       <Uri>https://dev.azure.com/dnceng/internal/_git/dotnet-runtime</Uri>
       <Sha>5535e31a712343a63f5d7d796cd874e563e5ac14</Sha>
```
Comment (rainersigwald) [inline eng/Version.Details.xml:58]: This needs updating too. From https://nuget.info/packages/System.Text.Json/8.0.3 it looks like it should be

```suggestion
      <Sha>9f4b1f5d664afdfc80e1508ab7ed099dff210fbd</Sha>
```

File: src/MSBuild/app.amd64.config [hunk 2]
```diff
@@ -134,8 +134,8 @@
         </dependentAssembly>
         <dependentAssembly>
           <assemblyIdentity name="System.Text.Json" publicKeyToken="cc7b13ffcd2ddd51" culture="neutral" />
-          <bindingRedirect oldVersion="0.0.0.0-8.0.0.0" newVersion="8.0.0.0" />
-          <codeBase version="8.0.0.0" href="..\System.Text.Json.dll"/>
+          <bindingRedirect oldVersion="0.0.0.0-8.0.3.0" newVersion="8.0.3.0" />
+          <codeBase version="8.0.3.0" href="..\System.Text.Json.dll"/>
```
Comment (rainersigwald) [inline src/MSBuild/app.amd64.config:138]:
```suggestion
          <bindingRedirect oldVersion="0.0.0.0-8.0.0.3" newVersion="8.0.0.3" />
          <codeBase version="8.0.0.3" href="..\System.Text.Json.dll"/>
```

File: src/MSBuild/app.config [hunk 3]
```diff
@@ -90,7 +90,7 @@
         </dependentAssembly>
         <dependentAssembly>
           <assemblyIdentity name="System.Text.Json" publicKeyToken="cc7b13ffcd2ddd51" culture="neutral" />
-          <bindingRedirect oldVersion="0.0.0.0-8.0.0.0" newVersion="8.0.0.0" />
+          <bindingRedirect oldVersion="0.0.0.0-8.0.3.0" newVersion="8.0.3.0" />
```
Comment (rainersigwald) [inline src/MSBuild/app.config:93]:
```suggestion
          <bindingRedirect oldVersion="0.0.0.0-8.0.0.3" newVersion="8.0.0.3" />
```

**Review by YuliiaKovalova — APPROVED** [review]

**Review by rainersigwald — APPROVED** [review]
Normally I'd want an exp/ insertion to validate the VS side of these changes but since they've already gone ahead I think it makes sense to take this now and try to insert into that PR.

**Comment by Forgind** [comment]
Does this break our S.T.J loading, or do we get binding redirected to the right version?
https://github.com/dotnet/sdk/pull/39901

**Comment by rainersigwald** [comment]
@Forgind https://github.com/dotnet/sdk/pull/39901#issuecomment-2030505355

**Pull request merged**